src/main/java/app/daos/impl/GameDAOImpl.java

```diff
@@ -28,6 +33,34 @@ public Game create(Game game) {
                 throw new EntityExistsException(String.format("Game with id %d already exists", game.getId()));
             }
 
+            if (game.getGenreSet() != null) {
+                Set<Genre> genreSet = new HashSet<>();
+                game.getGenreSet().forEach(genre -> {
+                    Genre foundGenre = em.find(Genre.class, genre.getId());
+
+                    if (foundGenre != null) {
+                        genreSet.add(foundGenre);
+                    } else {
+                        throw new EntityNotFoundException("Genre does not exist.");
+                    }
+                });
+                game.setGenreSet(genreSet);
+            }
+
+            if (game.getPlatformSet() != null) {
+                Set<Platform> platformSet = new HashSet<>();
+                game.getPlatformSet().forEach(platform -> {
+                    Platform foundPlatform = em.find(Platform.class, platform.getId());
+
+                    if (foundPlatform != null) {
+                        platformSet.add(foundPlatform);
+                    } else {
+                        throw new EntityNotFoundException("Platform does not exist.");
```

Comment:
The error message should include the platform id. for example "Platform with id %d could not be found"